modal/_container_entrypoint.py

```diff
@@ -282,7 +282,7 @@ async def _generate_inputs(self) -> AsyncIterator[tuple[str, str, api_pb2.Functi
                         yield (item.input_id, item.function_call_id, input_pb)
                         yielded = True
 
-                        if item.input.final_input:
+                        if item.input.final_input or self.function_def.reuse_containers is False:
```

Comment:
I think this will be true for all old clients?

Reply:
Oh yeah, I just realized a minute ago. Looks like we still need to negate this in function_def

Reply:
Actually, since the change is only in the client, old clients won't even have this condition in their source code, so it should still work

Reply:
I think that in this case, it could be cleaner to go for a `max_total_inputs_container` on the function definition where 0 is the default no limit.

User-facing field can still be `reuse_containers=False` (translate that to `max_total_inputs_container=1` in the definition). WDYT?

Reply:
I'm also fine with `reuse_containers=False` being a special case of `max_inputs=1`. @aksh-at actually [suggested](https://modal-com.slack.com/archives/C02TENSMNMQ/p1706988491710919) this initially, as a user-facing field. The concern at the time was implementation effort, but now we know that's not an issue.

Reply:
Oh right, if we can implement it client-side, `max_inputs` is better. The worker doesn't optimistically fetch more inputs right? I don't remember how that works, but that would break the "client-side only" implementation.

Reply:
Btw why `x is False` and not just `not x`. Its equivalent for booleans

Reply:
Btw I was stupid and didn't realize this code is in the client. Yeah it will work!

Reply:
> Btw why x is False and not just not x. Its equivalent for booleans

I wrote `x is False` back when I thought `x` could be `True`, `False`, or `None`, and only wanted `False`. `not x` would accept both `None` and `False`.

But this is no longer relevant now that we've switched to `max_inputs`, which isn't a boolean